x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -697,6 +703,27 @@ protected List<ReloadableSecurityComponent> getReloadableSecurityComponents() {
         return this.reloadableComponents.get();
     }
 
+    /*
+     * Copied from XPackPlugin.resolveConfigFile so we don't go to a different codesource
+     * and so fail the secured file permission check on the users file
+     */
+    public static Path resolveSecuredConfigFile(Environment env, String file) {
+        Path config = env.configFile().resolve(file);
```

Comment:
I would add a comment explaining that the resolved file does not need to be checked as rooted in the config directory because the secured config file permission check will already do that.